### PR TITLE
ThumbnailsTNG using SVG

### PR DESCRIPTION
**Pull request by phenaproxima**
So, this is kind of a coup-de-grace...

This PR introduces the ability to generate thumbnails for tweets which are purely textual (unaccompanied by images or other media). The thumbnails are SVGs containing the text of the tweet -- without hyperlinks or any other interactivity -- the name of the author, and their avatar. The SVG itself is generated using a Twig template and the theme system.

If adopted, this blows open the doors for what's possible in terms of thumbnail generation.

### CODE REVIEW

File: src/Plugin/MediaEntity/Type/Twitter.php [hunk 1]
```diff
@@ -163,6 +176,20 @@ public function getField(MediaInterface $media, $name) {
             return $tweet['retweet_count'];
           }
           return FALSE;
+
+        case 'thumbnail':
+          $uri = 'public://twitter-thumbnails/' . $matches['id'] . '.svg';
```
Comment (slashrsm) [inline src/Plugin/MediaEntity/Type/Twitter.php]: Would make sense to make this location configurable.

Comment (slashrsm) [inline src/Plugin/MediaEntity/Type/Twitter.php]: Would make sense to make this location configurable.

File: src/Plugin/MediaEntity/Type/Twitter.php [hunk 2]
```diff
@@ -163,6 +176,20 @@ public function getField(MediaInterface $media, $name) {
...
+          if (! file_exists($uri)) {
+            $build = [
+              '#theme' => 'media_entity_twitter_tweet_thumbnail',
+              '#tweet' => $tweet['text'],
+              '#author' => $tweet['user']['name'],
+              '#avatar' => $tweet['user']['profile_image_url'],
```
Comment (slashrsm) [inline src/Plugin/MediaEntity/Type/Twitter.php]: What happens if user changes profile image. Our SVG will still point to the old one, which will be gone at that point.

**Comment by slashrsm** [comment]
This is amazing feature. As I said, we should definitely implement this for Instagram too. Left two comments on diff.

**Comment by slashrsm** [comment]
Noticed two more things while testing:
- we should make sure directory exists and has sufficient permissions
- we should figure out how to handle display part (Drupal tries to display it with image style applied by default)

File: src/Plugin/MediaEntity/Type/Twitter.php (hunk 2 above)
Comment (phenaproxima) [inline src/Plugin/MediaEntity/Type/Twitter.php]: I don't have a good answer for this off the top of my head. It'd be expensive and extremely difficult (potentially impossible) to regenerate the thumbnails entirely every time they're viewed. So for the time being, I don't have any solution to this problem yet. (The username is embedded in the thumbnail, at least, and that won't change.)

**Comment by phenaproxima** [comment]
Regarding display -- I have a patch against core (still trying to convince people to go along with it, so +1s would be helpful!) for that. https://www.drupal.org/node/2652138

File: src/Plugin/MediaEntity/Type/Twitter.php (hunk 2 above)
Comment (slashrsm) [inline src/Plugin/MediaEntity/Type/Twitter.php]: Could we fetch image and store it locally?

Comment (phenaproxima) [inline src/Plugin/MediaEntity/Type/Twitter.php]: That might work. I'll see what I can do.

**Comment by slashrsm** [comment]
Latest patch from the issue was committed.

**Pull request closed without merging**